### DISCUSSION TITLE
Block coding beginner questions

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#1377] [Q&A]
Great opportunity, pybricks team!

1. Looking at the block-coding page on beta.pybricks.com I see these icons:
![afbeelding](https://github.com/pybricks/support/assets/8142081/2707099f-df85-437b-935f-58c1ae45856a)
Would it be possible to show a hint for all icons on mouse-hover?
That would not be possible on touchscreen devices, I think.
BTW the garbage bin seems to stick a bit to the left on this PC. (Nit-picking, sorry)

2. Have defined a function with an argument.
In that function, for debugging, I want to print the value of that argument.
But I cannot find how to do that. In other words: can not find a block that represents the argument.
Expected some oval block in "Variables" with a drop-down with argument names.

3. Missing the possibility of adding comments on the block code side.
I remember some item about this, but cannot find it now.

4. On the LEGO app(s) is a possibility to indicate a level of expertise that determines what block are shown.
Is something like that in your (may I say our) planning? I would only use "show all".
But I wondered.

Nothing with urgency in this item. 

Bert

**Comment by laurensvalk** [comment]
Thanks for the feedback!

For function arguments, you can just drag them to wherever you want to use them. (Just like in Scratch).

**Comment by BertLindeman** [comment]
> 
> For function arguments, you can just drag them to wherever you want to use them. (Just like in Scratch).

Hihi, Laurens, I'm an "Block coding beginner" Urhm, Scratch? 😜 
If you know, (and I do now) it is so simple. Just copy the argument(s) from the top of the function definition.
Thanks.

**Reply by abliss** [reply]
Thanks for this! Just FYI, I searched for more than 20 minutes before finding this. I had tried everything I could think of and was just about to submit a bugreport declaring it impossible. Nothing about the task's parameter bubble graphic affords dragging (though I am familiar with scratch)--none of the other bubbles that look similar are (afaik) draggable. Maybe it should have a handle? It would be great to add an example to https://pybricks.com/learn/flow-basics/tasks-functions/ , maybe with an animated gif showing how to drag the parameter out. (Or, maybe add a new block that can be found in one of the Tasks/Data/Variables drawer? AFAIK every other time the user wants to insert a value bubble, they have to get it from a drawer.)
